src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -777,17 +784,34 @@ public boolean isNullUnspecified() {
     }
 
     public boolean sameNullabilityAs(Type t) {
```

Comment:
Should these methods be here or in Types, since they are general relation on types?

Reply:
Also, looking at all these tests I wonder it we shouldn't maybe expose the NullMarker enum directly with a single accessor, and then use that (e.g. in a switch) to do the various tests. But this is just minor code style comments, no need to block this PR (mostly thinking aloud).